projects/plugins/jetpack/modules/contact-form/js/form-styles.js

```diff
@@ -0,0 +1,78 @@
+jQuery( function ( $ ) {
+	const EDITOR_SELECTOR = '[data-type="jetpack/contact-form"]';
+	const FRONTEND_SELECTOR = '.wp-block-jetpack-contact-form-container';
+
+	$( document ).ready( function () {
+		generateStyleVariables( FRONTEND_SELECTOR );
+		generateStyleVariables( EDITOR_SELECTOR );
+		observeContentChanges();
+	} );
+
+	function observeContentChanges() {
+		const observer = new MutationObserver( () => {
+			generateStyleVariables( FRONTEND_SELECTOR );
+			generateStyleVariables( EDITOR_SELECTOR );
+		} );
+
+		observer.observe( document.querySelector( 'body' ), {
+			childList: true,
+			subtree: true,
+		} );
+	}
+
+	function generateStyleVariables( selector, outputSelector = 'body' ) {
+		const STYLE_PROBE_CLASS = 'contact-form__style-probe';
+		const HTML = `
+			<div class="${ STYLE_PROBE_CLASS } contact-form" style="position: absolute; z-index: -1; width: 1px; height: 1px; visibility: hidden">
+				<div class="wp-block-button is-style-outline">
+					<div class="wp-block-button__link">Test</div>
+				</div>
+				<div class="jetpack-field">
+					<input class="components-text-control__input" type="text">
+				</div>
+			</div>
+		`;
+
+		if ( ! $( selector ).length || $( `.${ STYLE_PROBE_CLASS }` ).length ) {
+			return;
+		}
+
+		const outputContainer = $( outputSelector );
+		const container = $( selector );
+		const styleProbe = $( HTML );
+		const styleProbeEl = styleProbe[ 0 ];
+		container.append( styleProbe );
+
+		const bodyNode = document.querySelector( 'body' );
+		const buttonNode = styleProbeEl.querySelector( '.wp-block-button__link' );
+		const inputNode = styleProbeEl.querySelector( 'input[type="text"]' );
+
+		const primaryColor = window.getComputedStyle( buttonNode ).borderColor;
+		const backgroundColor = window.getComputedStyle( bodyNode ).backgroundColor;
+		const textColor = window.getComputedStyle( inputNode ).color;
+		const border = window.getComputedStyle( inputNode ).border;
+		const borderColor = window.getComputedStyle( inputNode ).borderColor;
+		const borderWidth = window.getComputedStyle( inputNode ).borderWidth;
+		const borderStyle = window.getComputedStyle( inputNode ).borderStyle;
+		const borderRadius = window.getComputedStyle( inputNode ).borderRadius;
+		const inputBackground = window.getComputedStyle( inputNode ).backgroundColor;
+		const inputPadding = window.getComputedStyle( inputNode ).padding;
+		const fontSize = window.getComputedStyle( inputNode ).fontSize;
+		const fontFamily = window.getComputedStyle( inputNode ).fontFamily;
+		const lineHeight = window.getComputedStyle( inputNode ).lineHeight;
```

Comment:
A bit of destructuring here would probably help make things more readable.

Reply:
Good suggestion, thanks!
Added some changes to the file.
Please let me know if there's something I could improve even better.